azurerm/internal/services/policy/data_source_policy_definition.go

```diff
@@ -124,3 +113,30 @@ func dataSourceArmPolicyDefinitionRead(d *schema.ResourceData, meta interface{})
 
 	return nil
 }
+
+func getPolicyDefinitionByDisplayName(ctx context.Context, client *policy.DefinitionsClient, displayName, managementGroupID string) (policy.Definition, error) {
+	var policyDefinitions policy.DefinitionListResultIterator
+	var err error
+
+	if managementGroupID != "" {
+		policyDefinitions, err = client.ListByManagementGroupComplete(ctx, managementGroupID)
+	} else {
+		policyDefinitions, err = client.ListComplete(ctx)
+	}
+	if err != nil {
+		return policy.Definition{}, fmt.Errorf("failed to load Policy Definition List: %+v", err)
+	}
+
+	for policyDefinitions.NotDone() {
```

Comment:
If we only find one match could we just error out?

Reply:
Done